File: app/src/main/java/com/itachi1706/busarrivalsg/CEPASScanActivity.java
package com.itachi1706.busarrivalsg;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.itachi1706.cepaslib.CEPASLibBuilder;
import com.itachi1706.cepaslib.app.feature.main.MainActivity;

public class CEPASScanActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        CEPASLibBuilder builder = CEPASLibBuilder.INSTANCE;
        builder.setPreferenceClass(MainSettings.class);
        builder.updateTitleBarColor(R.color.primary);
        builder.updateAccentColor(R.color.accent);
        builder.updateErrorColor(R.color.primaryDark);
        builder.setCustomTitle("Scan EZ-Link Card");
        builder.setHomeScreenWithBackButton(true);
        startActivity(new Intent(this, MainActivity.class));
        finish();
    }
}

File: app/src/main/java/com/itachi1706/busarrivalsg/CEPASScanActivity.kt
package com.itachi1706.busarrivalsg

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.itachi1706.cepaslib.CEPASLibBuilder
import com.itachi1706.cepaslib.app.feature.main.MainActivity

class CEPASScanActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val primaryColorRec = obtainStyledAttributes(intArrayOf(com.google.android.material.R.attr.colorPrimarySurface))
        val accentColorRec = obtainStyledAttributes(intArrayOf(com.google.android.material.R.attr.colorSecondary))
        val errorColorRec = obtainStyledAttributes(intArrayOf(com.google.android.material.R.attr.colorOnError))

        CEPASLibBuilder.apply {
            setPreferenceClass(MainSettings::class.java)
            updateTitleBarColor(primaryColorRec.getResourceId(0, 0))
            updateAccentColor(accentColorRec.getResourceId(0, 0))
            updateErrorColor(errorColorRec.getResourceId(0, 0))
            customTitle = "Scan EZ-Link Card"
            homeScreenWithBackButton = true
        }
        primaryColorRec.recycle()
        accentColorRec.recycle()
        errorColorRec.recycle()
        startActivity(Intent(this, MainActivity::class.java))
        finish()
    }
}